src/data/complex/module.lean

```diff
@@ -33,19 +34,49 @@ namespace complex
 
 instance algebra_over_reals : algebra ℝ ℂ := (complex.of_real).to_algebra
 
-@[simp] lemma coe_algebra_map : ⇑(algebra_map ℝ ℂ) = complex.of_real := rfl
+@[simp] lemma smul_coe {x : ℝ} {z : ℂ} : x • z = x * z := rfl
+
+section
+open_locale complex_order
+
+lemma complex_ordered_semimodule : ordered_semimodule ℝ ℂ :=
+{ smul_lt_smul_of_pos := λ z w x h₁ h₂,
+  begin
+    obtain ⟨y, l, rfl⟩ := lt_def.mp h₁,
+    refine lt_def.mpr ⟨x * y, _, _⟩,
+    exact mul_pos h₂ l,
+    ext; simp [mul_add],
+  end,
+  lt_of_smul_lt_smul_of_pos := λ z w x h₁ h₂,
+  begin
+    obtain ⟨y, l, e⟩ := lt_def.mp h₁,
+    by_cases h : x = 0,
+    { subst h, simp at h₁, exfalso, exact lt_irrefl 0 h₁, },
```

Comment:
Can you rephrase to avoid the nonterminal simp? Guessing `apply lt_irrefl 0, simpa using h1` or something will work?

Reply:
Fixed!